### ISSUE TITLE
I have two questions on DetectoRS implementation

### ISSUE THREAD

**Issue by grenoblien** [open-mmlab/mmdetection#3297]
**1) I have NaN loss on s0.loss and s1.loss.** 

What does s0, s1 loss mean?

loading annotations into memory...
Done (t=0.00s)
creating index...
index created!
2020-07-13 15:07:37,492 - mmdet - INFO - Start running, host: ubuntu@nipa2019-0012, work_dir: /home/common/Analysis/Detection/mmdetection/exp
2020-07-13 15:07:37,493 - mmdet - INFO - workflow: [('train', 1)], max: 12 epochs
2020-07-13 15:08:03,525 - mmdet - INFO - Epoch [1][50/1743]     lr: 1.978e-03, eta: 3:01:01, time: 0.521, data_time: 0.047, memory: 3869, loss_rpn_cls: 0.3856, loss_rpn_bbox: 0.0641, s0.loss_cls: 0.8218, s0.acc: 92.3301, s0.loss_bbox: 0.0145, s1.loss_cls: 0.4169, s1.acc: 90.0996, s1.loss_bbox: 0.0036, s2.loss_cls: 0.2579, s2.acc: 85.0879, s2.loss_bbox: 0.0007, loss: 1.9653
2020-07-13 15:08:25,403 - mmdet - INFO - Epoch [1][100/1743]    lr: 3.976e-03, eta: 2:46:11, time: 0.438, data_time: 0.004, memory: 3869, loss_rpn_cls: 19152174516294124.0000, loss_rpn_bbox: 14498697710162414.0000, s0.loss_cls: nan, s0.acc: 67.6660, s0.loss_bbox: nan, s1.loss_cls: nan, s1.acc: 71.0527, s1.loss_bbox: nan, s2.loss_cls: nan, s2.acc: 71.1250, s2.loss_bbox: nan, loss: nan
2020-07-13 15:08:47,342 - mmdet - INFO - Epoch [1][150/1743]    lr: 5.974e-03, eta: 2:41:09, time: 0.439, data_time: 0.004, memory: 3869, loss_rpn_cls: 0.6190, loss_rpn_bbox: 0.0519, s0.loss_cls: nan, s0.acc: 0.0000, s0.loss_bbox: nan, s1.loss_cls: nan, s1.acc: 0.0000, s1.loss_bbox: nan, s2.loss_cls: nan, s2.acc: 0.0000, s2.loss_bbox: nan, loss: nan
2020-07-13 15:09:07,909 - mmdet - INFO - Epoch [1][200/1743]    lr: 7.972e-03, eta: 2:36:04, time: 0.411, data_time: 0.004, memory: 3869, loss_rpn_cls: 0.5027, loss_rpn_bbox: 0.0362, s0.loss_cls: nan, s0.acc: 0.0000, s0.loss_bbox: nan, s1.loss_cls: nan, s1.acc: 0.0000, s1.loss_bbox: nan, s2.loss_cls: nan, s2.acc: 0.0000, s2.loss_bbox: nan, loss: nan
2020-07-13 15:09:28,540 - mmdet - INFO - Epoch [1][250/1743]    lr: 9.970e-03, eta: 2:32:59, time: 0.413, data_time: 0.004, memory: 3869, loss_rpn_cls: 0.4195, loss_rpn_bbox: 0.0408, s0.loss_cls: nan, s0.acc: 0.0000, s0.loss_bbox: nan, s1.loss_cls: nan, s1.acc: 0.0000, s1.loss_bbox: nan, s2.loss_cls: nan, s2.acc: 0.0000, s2.loss_bbox: nan, loss: nan
2020-07-13 15:09:49,180 - mmdet - INFO - Epoch [1][300/1743]    lr: 1.197e-02, eta: 2:30:49, time: 0.413, data_time: 0.004, memory: 3869, loss_rpn_cls: 0.3660, loss_rpn_bbox: 0.0442, s0.loss_cls: nan, s0.acc: 0.0000, s0.loss_bbox: nan, s1.loss_cls: nan, s1.acc: 0.0000, s1.loss_bbox: nan, s2.loss_cls: nan, s2.acc: 0.0000, s2.loss_bbox: nan, loss: nan
2020-07-13 15:10:09,829 - mmdet - INFO - Epoch [1][350/1743]    lr: 1.397e-02, eta: 2:29:10, time: 0.413, data_time: 0.004, memory: 3869, loss_rpn_cls: 0.3506, loss_rpn_bbox: 0.0507, s0.loss_cls: nan, s0.acc: 0.0000, s0.loss_bbox: nan, s1.loss_cls: nan, s1.acc: 0.0000, s1.loss_bbox: nan, s2.loss_cls: nan, s2.acc: 0.0000, s2.loss_bbox: nan, loss: nan
2020-07-13 15:10:30,343 - mmdet - INFO - Epoch [1][400/1743]    lr: 1.596e-02, eta: 2:27:45, time: 0.410, data_time: 0.004, memory: 3869, loss_rpn_cls: 0.3164, loss_rpn_bbox: 0.0433, s0.loss_cls: nan, s0.acc: 0.0000, s0.loss_bbox: nan, s1.loss_cls: nan, s1.acc: 0.0000, s1.loss_bbox: nan, s2.loss_cls: nan, s2.acc: 0.0000, s2.loss_bbox: nan, loss: nan
2020-07-13 15:10:50,859 - mmdet - INFO - Epoch [1][450/1743]    lr: 1.796e-02, eta: 2:26:33, time: 0.410, data_time: 0.004, memory: 3869, loss_rpn_cls: 0.2812, loss_rpn_bbox: 0.0371, s0.loss_cls: nan, s0.acc: 0.0000, s0.loss_bbox: nan, s1.loss_cls: nan, s1.acc: 0.0000, s1.loss_bbox: nan, s2.loss_cls: nan, s2.acc: 0.0000, s2.loss_bbox: nan, loss: nan
2020-07-13 15:11:11,489 - mmdet - INFO - Epoch [1][500/1743]    lr: 1.996e-02, eta: 2:25:37, time: 0.413, data_time: 0.004, memory: 3869, loss_rpn_cls: 0.2666, loss_rpn_bbox: 0.0342, s0.loss_cls: nan, s0.acc: 0.0000, s0.loss_bbox: nan, s1.loss_cls: nan, s1.acc: 0.0000, s1.loss_bbox: nan, s2.loss_cls: nan, s2.acc: 0.0000, s2.loss_bbox: nan, loss: nan

**2) I have added my own data set and which part should I change to train only on 4 classes?**

here is my config files.

I have not touched '../_base_/models/cascade_rcnn_r50_fpn.py' so I leave it unposted.

--------------------------------------

_base_ = [
    '../_base_/models/cascade_rcnn_r50_fpn.py',
    # '../_base_/datasets/coco_detection.py',
    '../_base_/datasets/tree_detection.py',

    '../_base_/schedules/schedule_1x.py', '../_base_/default_runtime.py'
]

model = dict(
    backbone=dict(
        type='DetectoRS_ResNet',
        conv_cfg=dict(type='ConvAWS'),
        sac=dict(type='SAC', use_deform=True),
        stage_with_sac=(False, True, True, True),
        output_img=True),
    neck=dict(
        type='RFP',
        rfp_steps=2,
        aspp_out_channels=64,
        aspp_dilations=(1, 3, 6, 1),
        rfp_backbone=dict(
            rfp_inplanes=256,
            type='DetectoRS_ResNet',
            depth=50,
            num_stages=4,
            out_indices=(0, 1, 2, 3),
            frozen_stages=1,
            norm_cfg=dict(type='BN', requires_grad=True),
            norm_eval=True,
            conv_cfg=dict(type='ConvAWS'),
            sac=dict(type='SAC', use_deform=True),
            stage_with_sac=(False, True, True, True),
            pretrained='torchvision://resnet50',
            style='pytorch')))

-----------------------------------------------

_base_ = 'tree_detection.py'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True, with_mask=True),
    dict(type='Resize', img_scale=(511, 511), keep_ratio=True),
    dict(type='RandomFlip', flip_ratio=0.5),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='Pad', size_divisor=32),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels']),
]
data = dict(train=dict(pipeline=train_pipeline))
evaluation = dict(metric=['bbox'])

---------------------------------------------------

dataset_type = 'TreeDataset'
data_root = 'data/tree/'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='Resize', img_scale=(511, 511), keep_ratio=True),
    dict(type='RandomFlip', flip_ratio=0.5),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='Pad', size_divisor=32),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels']),
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(511, 511),
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(type='Normalize', **img_norm_cfg),
            dict(type='Pad', size_divisor=32),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img']),
        ])
]
data = dict(
    samples_per_gpu=2,
    workers_per_gpu=2,
    train=dict(
        type=dataset_type,
        ann_file=data_root + 'annotations/train.json',
        img_prefix=data_root + 'train/',
        pipeline=train_pipeline),
    val=dict(
        type=dataset_type,
        ann_file=data_root + 'annotations/val.json',
        img_prefix=data_root + 'val/',
        pipeline=test_pipeline),
    test=dict(
        type=dataset_type,
        ann_file=data_root + 'annotations/test.json',
        img_prefix=data_root + 'val/',
        pipeline=test_pipeline))
evaluation = dict(interval=1, metric='bbox')

----------------------------------------------------

Many thanks in advance!!

**Comment by roadcode** [comment]
same as you. after train some step (about 400 step), the s0.loss_cls and s0.loss_bbox is nan in my own dataset.

**Comment by HaoweiGis** [comment]
I have the same question！

**Comment by ZwwWayne** [comment]
The related bug should have been fixed in https://github.com/open-mmlab/mmdetection/pull/3187. If you still find that the loss goes to be Nan, you can change the `warmup_iters` from 500 to 1000 or 2000. You can also add gradient clip to restrict the gradient. These two methods should help reduce the risk of loss going to be Nan.

**Comment by yutao007** [comment]
这个问题确实现在的版本2存在，换成版本1可以跑。

**Comment by hegee** [comment]
> 
> 
> 这个问题确实现在的版本2存在，换成版本1可以跑。

不好意思请问一下1版本的在哪呀？detectoRS不是2版本才开始支持的吗？
